imap_processing/codice/codice_l2.py

```diff
@@ -0,0 +1,74 @@
+"""
+Perform CoDICE l2 processing.
+
+This module processes CoDICE l1 files and creates L2 data products.
+
+Notes
+-----
+from imap_processing.codice.codice_l2 import process_codice_l2
+dataset = process_codice_l2(l1_filename)
+"""
+
+import logging
+from pathlib import Path
+
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.cdf.utils import load_cdf
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def process_codice_l2(file_path: Path) -> xr.Dataset:
+    """
+    Will process CoDICE l1 data to create l2 data products.
+
+    Parameters
+    ----------
+    file_path : pathlib.Path
+        Path to the CoDICE L1 file to process.
+
+    Returns
+    -------
+    l2_dataset : xarray.Dataset
+        The``xarray`` dataset containing the science data and supporting metadata.
+    """
+    logger.info(f"\nProcessing {file_path}")
+
+    # Open the l1 file
+    l1_dataset = load_cdf(file_path)
+
+    # Use the logical source as a way to distinguish between data products and
+    # set some useful distinguishing variables
+    dataset_name = l1_dataset.attrs["Logical_source"]
+    data_level = dataset_name.removeprefix("imap_codice_").split("_")[0]
+    descriptor = dataset_name.removeprefix(f"imap_codice_{data_level}_")
+    dataset_name = dataset_name.replace(data_level, "l2")
```

Comment:
Rather than all of this parsing I think you could take advantage of the objects in imap-data-access like `ScienceFilePath(file_path)` where you could then do something like: `filepath.descriptor`. Or maybe those are on ProcessingInput instead.